src/toil/jobStores/azureJobStore.py

```diff
@@ -278,9 +278,11 @@ def getSize(cls, url):
     @classmethod
     def _readFromUrl(cls, url, writable):
         blob = cls._parseWasbUrl(url)
-        blob.service.get_blob_to_file(container_name=blob.container,
-                                      blob_name=blob.name,
-                                      stream=writable)
+        for attempt in retry_azure():
+            with attempt:
+                blob.service.get_blob_to_file(container_name=blob.container,
```

Comment:
Thats a good point. I'm also not aware of a better solution for that behavior but it's a good thing to keep in mind if we do start seeing problems.